### PR TITLE
Expose more parameters in ModelConfig and add generatio

### PR DESCRIPTION
**Pull request by kira-lin**
Plus, replace use_auth_token with token

### CODE REVIEW

File: llm_on_ray/finetune/models/finetune_config_template.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ General:
   checkpoint_dir: /tmp/llm-ray/checkpoint
   config:
     trust_remote_code: false
-    use_auth_token: null
+    use_auth_tokenuth_token: null
```
Comment (carsonwang) [inline llm_on_ray/finetune/models/finetune_config_template.yaml:8]: Is this unintended modification?

File: tests/inference/test_utils.py [hunk 2]
```diff
@@ -57,10 +57,10 @@ def test_max_input_len():
 # Add more tests for edge cases
 
 
-def test_get_torch_dtype_cpu_without_ipex(mock_infer_conf):
-    hf_config = None
-    dtype = get_torch_dtype(mock_infer_conf, hf_config)
-    assert dtype == torch.get_default_dtype()
+def test_decide_torch_dtype_cpu_without_ipex(mock_infer_conf):
```
Comment (carsonwang) [inline tests/inference/test_utils.py:60]: can you please cover more test cases in your implementation, for example device is HPU?

File: llm_on_ray/inference/inference_config.py [hunk 3]
```diff
@@ -23,6 +23,8 @@ class ModelConfig(BaseModel):
     trust_remote_code: bool = False
     use_auth_token: Union[str, None] = None
     load_in_4bit: bool = False
+    torch_dtype: Union[str, None] = None
```
Comment (carsonwang) [inline llm_on_ray/inference/inference_config.py:26]: IPEX has a `precision` config, vLLM has a `precision` config too. Here we are adding a new one. All should use this single config, right? @kira-lin @KepingYan @xwu99

Comment (carsonwang) [reply llm_on_ray/inference/inference_config.py:26]: need to update the model yaml files.

Comment (kira-lin) [reply llm_on_ray/inference/inference_config.py:26]: this torch_dtype is used in AutoModelForCausalLM.from_pretrained. It's different from IPEX one. torch_dtype can be left empty to use default type.

This ModelConfig is just kwargs of from_pretrained function. Ideally, users should be able to specify any key supported by from_pretrained, but they can also run it without specifying anything.

**Review by carsonwang — APPROVED** [review]
Thanks!

**Comment by xwu99** [comment]
It's better to add logging info to show the decided dtype result. can use common/logging.py to do this.
